### DISCUSSION TITLE
elbow plot function

### DISCUSSION THREAD

**Discussion by mitchTwoTimes** [GreenleafLab/ArchR#1155] [Questions / Documentation] [answered]
hello,

Does archR support a way to generate elbow plots as a heuristic to determine the number of principal components? If not, are there any hacks to accomplish this?

Thank you!

**Comment by rcorces** [comment, marked as answer]
Are you asking whether ArchR provides a way to plot this information or are you asking whether ArchR gives you the information to be plotted? And by principal components, are you talking about LSI dimensions?

**Reply by mitchTwoTimes** [reply]
> Are you asking whether ArchR provides a way to plot this information or are you asking whether ArchR gives you the information to be plotted?

I suppose I'm asking both

> And by principal components, are you talking about LSI dimensions?

Oh, right. I'm referring to the `dimsToUse` arg in `addIterativeLSI`.

To rephrase, I would like to know if there is a method to settle on the number of dimensions to include when adding iterative LSI.

**Reply by rcorces** [reply]
This is definitely not a topic that I consider myself super well read on. That being said, I dont think you will find many reasons to exclude specific LSI dims. Certainly this is not very important during that `addIterativeLSI()` step but I also dont think there are many good reasons to change this from the default in downstream steps like `addUMAP()`.
ArchR automatically handles LSI dimensions that are correlated to sequencing depth (via the `corCutOff` parameter) so you dont need to worry about this like you would with other scATAC tools.

**Reply by mitchTwoTimes** [reply]
Thanks for the thoughtful response. In the context of multiome (paired) data, is it appropriate to include all dimensions for the gene expression matrix as well?

**Reply by rcorces** [reply]
sorry for the delay in replying. Yes - I would include all dims for RNA as well.

**Comment by AnjaliC4** [comment]
Hi @mitchTwoTimes, here is a simple way to make elbowPlots (as Signac does): 
rd <- getReducedDims(proj, reducedDims = "IterativeLSI",dimsToUse = 1:50)
sd <- colSds(rd)
ggplot(data.frame(dims = 1:50, stdev = sd[1:50]))+geom_point(mapping = aes_string(x = 'dims', y = 'stdev'))+labs(x = "LSI",y="Standard Deviation")

Hope it helps!

**Comment by DaianeH** [comment]
Hi,

I’m sending attached the elbow plot of LSI dimensions generated with the above code on my archR object, and it seems that the first dimension has the lowest SD - is it because it was correlated to depth so it was removed by archR?

Thank you in advance,

[elbow_estimatedLSI.pdf](https://github.com/GreenleafLab/ArchR/files/10795034/elbow_estimatedLSI.pdf)

**Comment by qdong2023** [comment]
Hi,

Is there a way to plot the correction of sequencing depth with each reduced dimension component in Archr? Like to function of DepthCor() in Signac?

Thanks
